src/OrchardCoreContrib.Linq/OrchardCoreDataContext.cs

```diff
@@ -0,0 +1,161 @@
+﻿using OrchardCore.AdminDashboard.Indexes;
+using OrchardCore.Alias.Indexes;
+using OrchardCore.AuditTrail.Indexes;
+using OrchardCore.ContentLocalization.Records;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Deployment.Indexes;
+using OrchardCore.Layers.Indexes;
+using OrchardCore.Lists.Indexes;
+using OrchardCore.OpenId.YesSql.Indexes;
+using OrchardCore.PublishLater.Indexes;
+using OrchardCore.Taxonomies.Indexing;
+using OrchardCore.Users.Indexes;
+using OrchardCore.Workflows.Indexes;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+using YesSql.Indexes;
+
+namespace OrchardCoreContrib.Linq
+{
+    /// <summary>
+    /// Represents a data context for Orchard Core index tables.
+    /// </summary>
+    public class OrchardCoreDataContext : IDisposable
+    {
+        private readonly IStore _store;
+
+        /// <summary>
+        /// Initializes a new instance of a <see cref="OrchardCoreDataContext"/> with store.
+        /// </summary>
+        /// <param name="store">The underlying store of the database.</param>
+        public OrchardCoreDataContext(IStore store)
+        {
+            _store = store;
+        }
+
+        /// <summary>
+        /// Gets a list of dashboards widgets metadata.
+        /// </summary>
+        public IEnumerable<DashboardPartIndex> Dashboards => QueryAsync<DashboardPartIndex>().Result;
```

Comment:
Why not use `async` methods instead, so `.Result` and the forced sync wouldn't be necessary?

Reply:
Properties can't be `async`

Reply:
Hence why I said `async` **methods** ;).

Reply:
Sorry I just focus on the last part ;) this is just an initial prototype, which can be enahnced as `IQueryable` provider to make similar to EF `DataContext`